src/main/java/seedu/lifeasier/parser/Parser.java

```diff
@@ -883,6 +894,122 @@ private boolean isNumeric(String input) {
         return true;
     }
 
+    /**
+     * Checks if the input module code is in a valid format.
+     *
+     * @param moduleCode The module code to be checked.
+     * @return true if the module code is determined to be valid.
+     */
+    public boolean checkIfValidModuleCode(String moduleCode) {
+        logger.log(Level.INFO, "Starting module code verification for: " + moduleCode);
+        moduleCode = moduleCode.trim();
+        char[] moduleCodeElements = moduleCode.toCharArray();
+
+        //Check if input for module code is within valid length of 8
+        if (moduleCodeElements.length > MAXIMUM_CODE_LENGTH || moduleCodeElements.length < MINIMUM_CODE_LENGTH) {
+            logger.log(Level.INFO, "Invalid module code length");
+            return false;
+        }
+
+        int letterCount = getLetterCount(moduleCodeElements);
+        //Check if module code prefix falls out of valid range of having 2-3 consecutive letters
+        if (letterCount < 2 || letterCount > 3) {
+            logger.log(Level.INFO, "Invalid prefix length");
+            return false;
+        }
+
+        int numberCount = getNumberCount(moduleCodeElements, letterCount);
+        //Check if there are 4 consecutive numbers in sequence
```

Comment:
You could consider renaming this such that the comment is not necessary.
```suggestion
        int consecutiveNumberCount = getConsecutiveNumberCount(moduleCodeElements, letterCount);
```

Reply:
Good suggestion, will change the one for letter count as well.